src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java

```diff
@@ -364,18 +366,51 @@ private String getAddress(Principal principal) {
     @RolesAllowed(RbacIdentityProvider.RBAC_READ_NOTIFICATIONS)
     public Uni<List<BehaviorGroup>> findBehaviorGroupsByBundleId(@Context SecurityContext sec, @PathParam("bundleId") UUID bundleId) {
         return getAccountId(sec)
-                .onItem().transformToUni(accountId -> behaviorGroupResources.findByBundleId(accountId, bundleId))
-                .onItem().call(behaviorGroups -> behaviorGroups.size() == 0 ? Uni.createFrom().voidItem() : Uni.combine().all().unis(
-                        behaviorGroups.stream()
-                                .map(BehaviorGroup::getActions)
-                                .map(bga -> bga.size() == 0 ? Uni.createFrom().voidItem() : Uni.combine().all().unis(
-                                        bga.stream()
-                                                .map(BehaviorGroupAction::getEndpoint)
-                                                .map(endpoint -> endpointResources.loadProperties(endpoint))
-                                                .collect(Collectors.toList())
-                                        )
-                                                .discardItems()
-                                ).collect(Collectors.toList())
-                ).discardItems());
```

Comment:
I didn't plan to change this at first but then a NPE showed up during tests and I ended up fixing other issues. This will be replaced soon by @josejulio's batch loading idea.